### PR TITLE
Influxdb: Response parser performance improvement

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**
Reduce the memory allocations and improve the speed in `influxdb`'s response parser.
BigJson is 15mb. 
```
❯ benchstat old.txt newest.txt
goos: darwin
goarch: arm64
pkg: github.com/grafana/grafana/pkg/tsdb/influxdb
                │   old.txt   │             newest.txt              │
                │   sec/op    │   sec/op     vs base                │
ParseJson-10      216.5µ ± 2%   147.2µ ± 0%  -32.01% (p=0.000 n=10)
ParseBigJson-10   766.1m ± 3%   530.6m ± 1%  -30.74% (p=0.000 n=10)
geomean           12.88m        8.837m       -31.38%

                │   old.txt    │              newest.txt              │
                │     B/op     │     B/op      vs base                │
ParseJson-10      199.3Ki ± 0%   150.9Ki ± 0%  -24.29% (p=0.000 n=10)
ParseBigJson-10   763.4Mi ± 0%   567.6Mi ± 0%  -25.65% (p=0.000 n=10)
geomean           12.19Mi        9.146Mi       -24.97%

                │   old.txt   │             newest.txt              │
                │  allocs/op  │  allocs/op   vs base                │
ParseJson-10      4.681k ± 0%   2.798k ± 0%  -40.23% (p=0.000 n=10)
ParseBigJson-10   17.39M ± 0%   10.42M ± 0%  -40.04% (p=0.000 n=10)
geomean           285.3k        170.8k       -40.13%
```

If you compare the old code and the new one, you can easily see the new code is ugly. But it does the job. 
If you want to run benchmark test on your own please see the comment in `response_parser_bench_test.go`

**Why do we need this feature?**

InfluxDB will parse the big json data even faster and use less memory allocation.

**Who is this feature for?**
All InfluxDB datasource users

### CODE REVIEW

File: pkg/tsdb/influxdb/response_parser_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+// go test -benchmem -run=^$ -memprofile memprofile.out -count=10 -bench ^BenchmarkParseJson$ github.com/grafana/grafana/pkg/tsdb/influxdb
+// go tool pprof -http=localhost:9999 memprofile.out
+func BenchmarkParseJson(b *testing.B) {
+	responseFileName := filepath.Join("./testdata", "response.json")
+
+	// nolint:gosec
+	// We can ignore the gosec G304 warning since this is a test file
+	responseBytes, err := os.ReadFile(responseFileName)
```
Comment (hairyhenderson) [inline pkg/tsdb/influxdb/response_parser_bench_test.go:18]: Can you also change the test to not read the file? (I sent you a patch in Slack for that 😅)

Comment (itsmylife) [reply pkg/tsdb/influxdb/response_parser_bench_test.go]: I've missed that patch. Now it is being embedded https://github.com/grafana/grafana/pull/64776/commits/a921188bdb2283be024849e261eae46bebd11164

**Comment by itsmylife** [comment]
Using `json-iterator` package results:
```
❯ benchstat newer.txt newest.txt
goos: darwin
goarch: arm64
pkg: github.com/grafana/grafana/pkg/tsdb/influxdb
                │  newer.txt  │             newest.txt              │
                │   sec/op    │   sec/op     vs base                │
ParseJson-10      151.6µ ± 0%   124.8µ ± 0%  -17.72% (p=0.000 n=10)
ParseBigJson-10   562.2m ± 2%   437.7m ± 1%  -22.15% (p=0.000 n=10)
geomean           9.233m        7.390m       -19.96%

                │  newer.txt   │             newest.txt              │
                │     B/op     │     B/op      vs base               │
ParseJson-10      150.9Ki ± 0%   140.7Ki ± 0%  -6.76% (p=0.000 n=10)
ParseBigJson-10   567.6Mi ± 0%   522.8Mi ± 0%  -7.90% (p=0.000 n=10)
geomean           9.146Mi        8.476Mi       -7.33%

                │  newer.txt   │             newest.txt              │
                │  allocs/op   │  allocs/op   vs base                │
ParseJson-10       2.798k ± 0%   2.102k ± 0%  -24.87% (p=0.000 n=10)
ParseBigJson-10   10.424M ± 0%   7.880M ± 0%  -24.41% (p=0.000 n=10)
geomean            170.8k        128.7k       -24.64%

```

File: pkg/tsdb/influxdb/response_parser.go [hunk 2]
```diff
@@ -11,6 +10,7 @@ import (
 
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+	json "github.com/json-iterator/go"
```
[the remark this answers is not in the document]
Comment (itsmylife) [reply pkg/tsdb/influxdb/response_parser.go:13]: That makes sense. Good point. I will change that.

[the remark this answers is not in the document]
Comment (itsmylife) [reply pkg/tsdb/influxdb/response_parser.go]: https://github.com/grafana/grafana/pull/64776/commits/591a28c43f8142044b0bf48388f59f112841be9e

**Review by hairyhenderson — APPROVED** [review]
LGTM! Thanks for this @itsmylife 🙇‍♂️

**Pull request merged**